### PR TITLE
Automate project versioning with semantic-release

### PR DESCRIPTION
**Pull request by matchai**
## Description

To reduce the amount of manual effort involved with versioning a release, this PR adds semantic-release to our project. In order to ensure that releases are versioned correctly, maintainers will be using the [Conventional Commits spec](https://www.conventionalcommits.org/en/v1.0.0-beta.2/) to standardize commit messages when squashing features branches onto master.

## Motivation and Context

Replaces the need to create version-incrementing commits to master.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

- [ ] I have tested using **MacOS**
- [ ] I have tested using **Linux**

## Checklist:

- [x] I have checked that no other PR duplicates mine
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,21 +6,38 @@
...
+    "docs:serve": "npm run docs:prepare && gitbook serve .",
     "contributors:add": "all-contributors add",
     "contributors:generate": "all-contributors generate"
   },
   "license": "MIT",
   "devDependencies": {
+    "@semantic-release/exec": "^3.3.1",
+    "@semantic-release/git": "^7.0.6",
     "all-contributors-cli": "^5.4.1",
-    "gitbook-cli": "^2.3.2",
```
Comment (matchai) [inline package.json:19]: It appears these gitbook dependencies were not needed in the first place, and are downloaded by gitbook when `gitbook install` is run.

File: .travis.yml [hunk 2]
```diff
@@ -16,3 +16,7 @@ jobs:
       install:
        - brew update
        - brew install fish
+    - stage: release
+      provider: script
+      skip_cleanup: true
+      script: npx travis-deploy-once "npx semantic-release"
```
Comment (Snuggle) [inline .travis.yml:22]: What is npx? This isn't a typo of npm, is it?

**Comment by Snuggle** [comment]
What is the purpose of this merge and is there going to be a change to how we should write our commit summaries? Should I read through the Convention Commits specification?

If version incrementing commits are no longer necessary, how would we actually increment the version? Is it done automatically? Under what criteria is it automatically incremented? How about the funny fish themed names?

File: .travis.yml (hunk 2 above)
Comment (matchai) [reply .travis.yml:22]: Nope, `npx` is a feature included in npm v5.
It'll download and cache the package for local use without having to install it as a dependency to a project. This will execute the binaries directly for our dependencies so that we don't have to provide the full path to them.

**Comment by Snuggle** [comment]
To add to this, could there please be some guidelines for collaborators such as myself? I'm presuming you'd like to leave the master branch alone and work entirely using merge/pull requests. Are we okay to squash-merge simple pull requests if everything is in order to lessen your load? I'm also guessing you'll always be the only person to increment the version number, since I can't imagine anyone else coming up with such awesome alliterative names. Would you like collaborators to review other's pull requests? Apply appropriate labels? I'd love to be clear and explicit, I don't want to step on your toes! 😊

**Comment by matchai** [comment]
This merge will make it so that moving forward, spacefish will be automatically versioned. I would suggest you read Conventional Commits spec, as it's not too long.

The version of spacefish should increment on its own, and GitHub releases will automatically be created. With this in place, there should be no issues with having other maintainers merge in requests, as long as they ensure that the squashed commit message fits the Conventional Commits standards.

Unfortunately, this will mean that we will no longer have fun names, unless we manually go back and put them in 😜 

In short, based on the commit message prefix, the version will be incremented appropriately:

**patch**
_1.0.0 -> 1.0.1_
```
fix: use only the first battery value for acpi
```

**minor**
_1.0.0 -> 1.1.0_
```
feat: add rust version section
```

**major**
_1.0.0 -> 2.0.0_
```
fix: remove poorly performing sections

BREAKING CHANGE: the git, package and node sections have been removed
```

**Comment by Snuggle** [comment]
Fancy as heck! This needs documentation for sure. Could we please have the guidelines updated and expanded upon, with the answers to the above questions as well, in the [CONTRIBUTING.md](https://github.com/matchai/spacefish/blob/master/CONTRIBUTING.md)?

~~Does this mean the funny fish names are over? :sob: :fish:~~

**Comment by matchai** [comment]
The reason I was thinking we didn't really need to add it to CONTRIBUTING.md was that it doesn't really need to be enforced by contributors, and only maintainers.

Do you still think it's worth documenting? 🤔

I'll probably still be giving releases names if I've got time to waste. 😛

**Review by Snuggle — APPROVED** [review]
I can't see anything glaringly wrong with this PR. As long as it works, I'd be happy to see these changes take place. :blush:

And perhaps it could go in a seperate MAINTAINERS.md or under a seperate section in CONTRIBUTING.md? Just somewhere that will be out of the way but still explicit and useful!

Spacefish is acquiring awesome astro-age automation! :robot::fish:

**Comment by matchai** [comment]
This isn't really something we can test until we see it run through Travis. 😅 
Here goes nothing! 🚀
